src/Exception/EnvException.php

```diff
@@ -13,10 +13,22 @@ class EnvException extends InvalidArgumentException
      */
     public static function invalidFile($path)
     {
-        return new static(sprintf(
-            'Environment file `%s` does not exist or is not readable',
-            $path
-        ));
+        $messages = [];
+        $filepath = realpath($path);
+
+        if (!$filepath) {
+            $messages[] = 'does not exist';
+        }
+
+        if ($filepath && !is_file($filepath)) {
+            $messages[] = 'exists and is not a file';
+        }
+
+        if ($filepath && !is_readable($filepath)) {
+            $messages[] = 'is not readable';
+        }
```

Comment:
I don't agree with this change. It puts too much logic into the exception factory.